### PR TITLE
Cleaned up create-app deprecations

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

Currently running `yarn backstage-cli repo list-deprecations` with a fresh run of `npx @backstage/create-app@latest` will result in files that are using deprecated code but have new locations where they should be imported from. This PR addresses that

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/create-app/templates/default-app/packages/backend/src/index.ts [hunk 1]
```diff
@@ -78,13 +77,13 @@ async function main() {
...
-  const techdocsEnv = useHotMemoize(module, () => createEnv('techdocs'));
-  const searchEnv = useHotMemoize(module, () => createEnv('search'));
-  const appEnv = useHotMemoize(module, () => createEnv('app'));
+  const catalogEnv = createEnv('catalog');
+  const scaffolderEnv = createEnv('scaffolder');
+  const authEnv = createEnv('auth');
+  const proxyEnv = createEnv('proxy');
+  const techdocsEnv = createEnv('techdocs');
+  const searchEnv = createEnv('search');
+  const appEnv = createEnv('app');
```
Comment (Rugvip) [inline packages/create-app/templates/default-app/packages/backend/src/index.ts:86]: Let's keep these in place for now, it doesn't really make sense to remove them while we're still shipping the old system as default

**Review by Rugvip — COMMENTED** [review]
Nice, thank you! 😁

File: packages/create-app/templates/default-app/packages/backend/src/plugins/search.ts.hbs [hunk 2]
```diff
@@ -54,8 +53,6 @@ export default async function createPlugin(
   const { scheduler } = await indexBuilder.build();
   scheduler.start();
 
-  useHotCleanup(module, () => scheduler.stop());
```
Comment (freben) [inline packages/create-app/templates/default-app/packages/backend/src/plugins/search.ts.hbs:57]: same for this one, let's keep it for now right?

Comment (awanlin) [reply packages/create-app/templates/default-app/packages/backend/src/plugins/search.ts.hbs]: Done 👍

File: packages/create-app/templates/default-app/packages/backend/src/index.ts (hunk 1 above)
Comment (awanlin) [reply packages/create-app/templates/default-app/packages/backend/src/index.ts]: Done 👍

**Comment by awanlin** [comment]
Ready for another look, please @freben

**Review by freben — APPROVED** [review]

**Comment by awanlin** [comment]
Thanks @freben 🚀

**Pull request merged**